mautic/mautic
# Features

| Q                                      | A
| -------------------------------------- | ---
| Branch?                                | features
| Bug fix?                               | yes
| New feature?                           | no
| Deprecations?                          | no
| BC breaks?                             | no
| Automated tests included?              | no
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 




#### Description:
Webhook/Http/Client->post() requires 3 arguments, but 4 was given, pushing the last argument (secret) out of bounds, resulting in a wrong signature when using the Webhook test send feature.

1. Go to Webhooks
2. Create a webhook (or open an existing one). Make sure a secret is present
3. Send test webhook compare the webhook-signature header at the consuming end with what the value is expected to be. They will not match. If at the consuming end you match the value of the signature using an empty string as a secret, it will match.

#### Steps to test this PR:
1. Load up [this PR](https://mautibox.com)
2. Go to Webhooks
3. Create a webhook to an existing consumer, make sure the secret is not empty and matches the value expected at the consuming end
4. Test that the webhook is successfully consumed, that the webhook-signature header matches the expected value.
